### PR TITLE
Migrate project-level Gradle build script to Kotlin DSL

### PR DESCRIPTION
**Pull request by thunderbiscuit**
## Description
WIP. Migrating the Gradle build scripts to Kotlin DSL.

## Notes to the reviewers
I have not tested the publishing workflow.

## All Submissions:
* [x] I've signed all my commits

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
I have migrated all 3 build scripts. A few notes:
1. I'm keen on Gradle and over time I want to make sure I understand all our plugins and the configurations we use, as well as how we can make our build scripts better. This PR however is just a straight port of the Groovy DSL scripts.
2. I removed the `withJavadocJar()` config in the `jvm` build file because it currently only produced an empty jar.
3. I have not tested the nexus publishing task, but I did test the publishToLocalMaven task for the Android library and it worked well.
4. @notmandatory can I ask how you currently pass the username and password through the `findProperty()` method? I just want to make sure I find it using my `val ossrhUsername: String? by project` line. (aka is it defined in your home `gradle.properties`, or as a environment variable, or passed as a command line argument? Reading the [docs](https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html#findProperty-java.lang.String-) I think it could come from many places. I'm also looking at [this article](https://tomgregory.com/gradle-project-properties-best-practices/) to try and figure out how to do it best.

**Comment by notmandatory** [comment]
I'm currently using `gradle.properties` to get the publishing authentication info.  Some more details here: https://github.com/bitcoindevkit/bdk-kotlin#how-to-publish

**Comment by notmandatory** [comment]
I merged the two pending PRs so now this one needs a rebase. Then I'll give it a test and otherwise LGTM.

**Comment by thunderbiscuit** [comment]
It was a weird rebase but I _think_ I got it. The issue was that you had made changes to files I had deleted, so I just needed to delete them again.

File: jvm/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+}
+
+dependencies {
+    implementation(platform("org.jetbrains.kotlin:kotlin-bom"))
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk7")
+    implementation("net.java.dev.jna:jna:5.8.0")
+    api("org.slf4j:slf4j-api:1.7.30")
+    testImplementation("junit:junit:4.13.2")
+    testRuntimeOnly("org.junit.vintage:junit-vintage-engine:5.8.2")
```
Comment (notmandatory) [inline jvm/build.gradle.kts:35]: For some reason I had to add this dependency for the `./gradlew :jvm:test` command to work.

**Review by notmandatory — APPROVED** [review]
ACK 933af8c70633b754f0967cdd26c87101d9916303

I tested a staging publish and it worked so a full (non-test) publish should work too.

File: jvm/build.gradle.kts (hunk 1 above)
Comment (thunderbiscuit) [reply jvm/build.gradle.kts:35]: Same with the migration to the new API! This PR was initially made against the older version of bdk-ffi and so it was working, but when I worked on updating the tests I also found I needed to add this `junit-vintage-engine`. Thanks for fixing!

**Pull request merged**